pallets/asset/src/lib.rs

```diff
@@ -1502,6 +1616,9 @@ impl<T: Trait> Module<T> {
             .checked_add(&value)
             .ok_or(Error::<T>::BalanceOverflow)?;
 
+        let ops: Vec<_> = iter::repeat(ProtocolOp::AssetCheckpoint).take(2).collect();
+        T::ProtocolFee::charge_fees(ops.as_ref())?;
```

Comment:
```suggestion
        T::ProtocolFee::charge_fees(&[ProtocolOp::AssetCheckpoint, ProtocolOp::AssetCheckpoint])?;
```
Simpler & no allocation.